### PR TITLE
Remove unused build options

### PR DESCRIPTION
**Pull request by hainest**
We should revisit using CPack.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -163,78 +163,6 @@ endif()
 set (VERSION_STRING "${DYNINST_MAJOR_VERSION}.${DYNINST_MINOR_VERSION}.${DYNINST_PATCH_VERSION}")
 set (DYNINST_NAME "DyninstAPI-${VERSION_STRING}")
 
-if(BUILD_TARBALLS)
```
Comment (vsoch) [inline CMakeLists.txt:166]: This giant big beautiful block of red deletion is very satisfying, +1 destroy it.

File: cmake/options.cmake [hunk 2]
```diff
@@ -7,11 +7,8 @@ option (LIGHTWEIGHT_SYMTAB "Use lightweight symtab interface for ParseAPI, ProcC
 # Use ParseAPI analysis in Stackwalker?
 option (SW_ANALYSIS_STEPPER "Use ParseAPI-based analysis stepper in Stackwalker" ON)
 
-option (BUILD_TARBALLS "Build Dyninst package tarballs. Requires git archive, tar, gzip." OFF)
 option (BUILD_RTLIB_32 "Build 32-bit runtime library on mixed 32/64 systems" OFF)
 
-option(BUILD_DOCS "Build manuals from LaTeX sources" ON)
```
Comment (vsoch) [inline cmake/options.cmake:13]: Would be nice to someday have web docs because reading pdfs is sooooo 2009.

**Review by kupsch — COMMENTED** [review]
Removing BUILD_TARBALLS is good as it is unused.

The BUILD_DOCS currently is broken; `make docs` sort of works the first time, but is broken for subsequent invocations.  It would be nice to have a make target to build the latex docs that had proper dependencies so it would rebuild correctly. If this is removed, there needs to be instructions on how build the docs added to the repo and perhaps a script to create the pdfs.

**Comment by vsoch** [comment]
I'm pretty good at that, I could probably help in a week or so, have too many "after work personal projects" right now but probably that won't be the case after I finish some up! I've done readthedocs with rst a gazillion times.

**Comment by hainest** [comment]
@vsoch I wasn't hinting- I promise. :smiley: If you have the cycles and will, I certainly won't turn away the help, though.

**Comment by vsoch** [comment]
yeah I can add to my list of "stuff to do" I'll get to it sooner than later. :)

**Comment by kupsch** [comment]
@hainest I understand that the latex documentation is only part of the documentation, but it wasn't clear to me how someone that forked the repo would know how to build the latex documentation; it would require them to know the existence of the google doc. If we are moving from latex soon then this is OK and the merge can proceed.  If not, then putting the instructions in README.md (or similar) that is versioned in the repo would be useful.

**Comment by hainest** [comment]
@kupsch Oh, sorry, I didn't get that part of what you were saying. As far as I know, no one but us has ever tried to build the documentation. Even the Red Hat folks use the generated PDFs. I'm pretty sure the BUILD_DOCS was a holdover from the autotools->CMake conversion. I would _really_ like to see us move to readthedocs soon. Creating the documentation PDFs is by far the biggest pain of doing a release. Since Vanessa offered a bit of her time, I think the two of us can put our heads together to do a bulk export. There are still doc bugs that need to be fixed (#800, #711, #394), but we can do that afterward.

**Pull request merged**